File: src/javaFiles/EnumMain.java
package javaFiles;

public class EnumMain {

    public static void main(String[] args) {
        ParticipantType type = ParticipantType.AGED;
        showPrice(type);
    }

    protected enum ParticipantType {
        STUDENT(900),
        NORMAL(1200),
        AGED(800);

        private int price;

        ParticipantType(int price) {
            this.price = price;
        }
    }

    private static void showPrice(ParticipantType type) {
        switch (type) {
            case STUDENT:
                System.out.printf("入館料金は%s円です", ParticipantType.STUDENT.price);
            case NORMAL:
                System.out.printf("入館料金は%s円です", ParticipantType.NORMAL.price);
            case AGED:
                System.out.printf("入館料金は%s円です", ParticipantType.AGED.price);
        }
    }
}

File: src/ktFiles/EnumMain.kt
package ktFiles


enum class ParticipantType(val price: Int) {
    STUDENT(900),
    NORMAL(1200),
    AGED(800)
}

fun main() {
    showPrice(ParticipantType.STUDENT)
}

private fun showPrice(type: ParticipantType) {
    // when
    when (type) {
        ParticipantType.STUDENT -> println("入館料金は${type.price}円です")
        ParticipantType.NORMAL -> println("入館料金は${type.price}円です")
        ParticipantType.AGED -> println("入館料金は${type.price}円です")
    }
}
